### DISCUSSION TITLE
How to Cancel in-progress conversion

### DISCUSSION THREAD

**Discussion by zlepper** [dlemstra/Magick.NET#992] [Help]
I have this pretty simple code:
```cs
protected override async Task Execute(Stream stream, ImageFormat format, Stream outputStream,
            CancellationToken cancellationToken)
        {
            using var image = new MagickImage(stream);

            var size = new MagickGeometry(format.Width, format.Height);

            image.Resize(size);

            await image.WriteAsync(outputStream);
        }
```

As can be seen in the method signature i accept a CancellationToken, but i don't see a way of actually either passing it to Magick actively canceling either the write or resize operations. I can always do a check between the different method calls, but that can still be many seconds between if i'm resizing a large image. Is there a way to cancel the job?

**Comment by dlemstra** [comment]
I could add a `CancellationToken` overload to the `WriteAsync` method but part of the operation will run synchronous. And for the other opration you might want to take a look at the `.Progress` event of the image. You could use that to cancel the resize operation.

**Reply by zlepper** [reply]
Yeah, i'm mostly interested in just stopping the entire operation, so the `.Progress` thing should work well enough for that. 

A cancellationtoken on write async would also be a nice thing, if possible, just to stop the writing. 


Sidenote: That progress event is just a thing i didn't know i wanted also :D

**Reply by dlemstra** [reply]
The next rlease will include extra overloads that will add support for using a `CancellationToken `.
